apps/builder/app/builder/features/ai/hooks/effect-event.ts

```diff
@@ -0,0 +1,23 @@
+import { useRef, useInsertionEffect, useCallback } from "react";
+
+const useSafeInsertionEffect =
+  typeof window !== "undefined"
+    ? useInsertionEffect
+    : (fn: () => void) => {
+        fn();
+      };
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export const useEffectEvent = <T extends Function>(callback?: T) => {
+  const ref = useRef(callback);
+
+  useSafeInsertionEffect(() => {
+    ref.current = callback;
+  });
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why does it fail? Refs can be excluded from deps. Linter knows this.

Reply:
<img width="919" alt="image" src="https://github.com/webstudio-is/webstudio/assets/5077042/f939dfda-e8b6-433a-b399-a6fe7a4778a6">